bgpd/bgp_updgrp_packet.c

```diff
@@ -462,23 +473,24 @@ bpacket_reformat_for_peer (struct bpacket *pkt, struct peer_af *paf)
                    (bgp_multiaccess_check_v4 (v4nh, peer) == 0) &&
                    !CHECK_FLAG(vec->flags,
                                BPKT_ATTRVEC_FLAGS_RMAP_NH_UNCHANGED) &&
-                   !peer_af_flag_check (peer, paf->afi, paf->safi,
+                   !peer_af_flag_check (peer, nhafi, paf->safi,
                                          PEER_FLAG_NEXTHOP_UNCHANGED))
             {
+              /* NOTE: not handling case where NH has new AFI */
                mod_v4nh = &peer->nexthop.v4;
                nh_modified = 1;
             }
 
-          if (nh_modified)
-            stream_put_in_addr_at (s, vec->offset + 1, mod_v4nh);
+          if (nh_modified)      /* allow for VPN RD */
+            stream_put_in_addr_at (s, vec->offset + 1 + nhlen - 4, mod_v4nh);
 
           if (bgp_debug_update(peer, NULL, NULL, 0))
-            zlog_debug ("u%" PRIu64 ":s%" PRIu64 " %s send UPDATE w/ nexthop %s",
+            zlog_debug ("u%" PRIu64 ":s%" PRIu64 " %s send UPDATE w/ nexthop %s%s",
                     PAF_SUBGRP(paf)->update_group->id, PAF_SUBGRP(paf)->id,
```

Comment:
We have two choices here, either make the + line line up with the PAF_SUBGRP.. line or move the PAF_SUBGRP... line line up with the correct spot for zlog_debug.  Adding a code change where we now have two separate indentations breaks the rule of 'do what the code does around me' that we try to have.  Of the two I would prefer fixing the indentation right.  The point of not changing lines is that it makes it harder to merge.  But since we are already changing the line above and line below the PAF_SUBGRP line it will kill any merges there.  As such choose one indentation for the zlog_debug and use it.